geopandas/geodataframe.py

```diff
@@ -578,11 +578,28 @@ def from_postgis(
 
         Examples
         --------
+        PostGIS
+
+        >>> from sqlalchemy import create_engine  # doctest: +SKIP
+        >>> db_connection_url = "postgres://myusername:mypassword@myhost:5432/mydb"
+        >>> con = create_engine(db_connection_url)  # doctest: +SKIP
         >>> sql = "SELECT geom, highway FROM roads"
+        >>> df = geopandas.GeoDataFrame.from_postgis(sql, con)  # doctest: +SKIP
 
         SpatiaLite
+
         >>> sql = "SELECT ST_Binary(geom) AS geom, highway FROM roads"
         >>> df = geopandas.GeoDataFrame.from_postgis(sql, con)  # doctest: +SKIP
+
+        The recommended method of reading from PostGIS is
+        :func:`geopandas.read_postgis`:
```

Comment:
If we really want to recommend this (and I think we should), we should maybe not include any example at all here (as it will only prompt people to copy it), but simply point to `read_postgis`, and ensure we have good examples there?

Reply:
This pattern mirrors what we agreed on in `from_file` docstring - https://geopandas.readthedocs.io/en/latest/docs/reference/api/geopandas.GeoDataFrame.from_file.html#geopandas.GeoDataFrame.from_file. I wanted to be consistent with that.